adapter.go

```diff
@@ -360,9 +375,94 @@ func (a *Adapter) Close() error {
 	return nil
 }
 
-// getTableInstance return the dynamic table name
-func (a *Adapter) getTableInstance() *CasbinRule {
-	return &CasbinRule{}
+// getTableObject return the dynamic table object
+func (a *Adapter) getTableObject() interface{} {
+	if a.customTable == nil {
+		return &CasbinRule{}
+	}
+	return a.customTable
+}
+
+func convertToTableDefault(lines interface{}) *[]CasbinRule {
+	cusLines := reflect.ValueOf(lines)
+	if cusLines.Kind() == reflect.Ptr {
+		cusLines = cusLines.Elem()
+	}
+
+	length := cusLines.Len()
+	retLines := make([]CasbinRule, length)
+
+	for i := 0; i < length; i++ {
+		tmp := cusLines.Index(i)
+
+		retLines[i].ID = uint(tmp.FieldByName("ID").Uint())
+		retLines[i].Ptype = tmp.FieldByName("Ptype").String()
+		retLines[i].V0 = tmp.FieldByName("V0").String()
+		retLines[i].V1 = tmp.FieldByName("V1").String()
+		retLines[i].V2 = tmp.FieldByName("V2").String()
+		retLines[i].V3 = tmp.FieldByName("V3").String()
+		retLines[i].V4 = tmp.FieldByName("V4").String()
+		retLines[i].V5 = tmp.FieldByName("V5").String()
+		retLines[i].V6 = tmp.FieldByName("V6").String()
+		retLines[i].V7 = tmp.FieldByName("V7").String()
+	}
+
+	return &retLines
+}
+
+func convertToTableCustomized(t interface{}, lines *[]CasbinRule) interface{} {
+	modelType := reflect.TypeOf(t)
+	if modelType.Kind() == reflect.Ptr {
+		modelType = modelType.Elem()
+	}
+	cusLines := reflect.New(reflect.ArrayOf(len(*lines), modelType))
+
+	for i, line := range *lines {
+		tmp := reflect.New(modelType).Elem()
+
+		tmp.FieldByName("ID").SetUint(uint64(line.ID))
+		tmp.FieldByName("Ptype").SetString(line.Ptype)
+		tmp.FieldByName("V0").SetString(line.V0)
+		tmp.FieldByName("V1").SetString(line.V1)
+		tmp.FieldByName("V2").SetString(line.V2)
+		tmp.FieldByName("V3").SetString(line.V3)
+		tmp.FieldByName("V4").SetString(line.V4)
+		tmp.FieldByName("V5").SetString(line.V5)
+		tmp.FieldByName("V6").SetString(line.V6)
+		tmp.FieldByName("V7").SetString(line.V7)
+
+		cusLines.Elem().Index(i).Set(tmp)
+	}
+
+	return cusLines.Interface()
+}
+
+func (a *Adapter) dbDelete(db *gorm.DB, conds ...interface{}) error {
+	return db.Delete(a.getTableObject(), conds...).Error
+}
+
+func (a *Adapter) dbFind(db *gorm.DB, linesPtr *[]CasbinRule, conds ...interface{}) error {
+	modelType := reflect.TypeOf(a.getTableObject())
```

Comment:
if modelType is CasbinRule, we can query directly without having to go through the next steps